pyccl/halos/massdef.py

```diff
@@ -265,7 +272,43 @@ class MassDefVir(MassDef):
     Args:
         c_m (string): concentration-mass relation.
     """
+    name = 'vir'
+
     def __init__(self, c_m='Klypin11'):
         super(MassDefVir, self).__init__('vir',
                                          'critical',
                                          c_m_relation=c_m)
+
+
+def mass_def_from_name(name):
+    """ Return mass definition subclass from name string.
+
+    Args:
+        name (string): a mass function name
+
+    Returns:
+        MassDef subclass corresponding to the input name.
+    """
+    if name == "vir":
+        c_m_relation = "Klypin11"
+        return MassDefVir
+
+    c_m_relation = "Duffy08"
```

Comment:
I am not sure I understand the logic of the lines below this one. I mean, I understand what you're doing, but it won't work in general, since `Duffy08` is only defined for `Delta=200`. I am not sure it is worth the extra code to have this function.

Reply:
OK. Indeed, I've now simplified the function. Simple is always better.